From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Sat, 22 Oct 2022 14:48:49 +0900
Subject: Use IPv6 address for content server default value

Forwarded: https://github.com/kovidgoyal/calibre/pull/2119
---
 src/calibre/srv/opts.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/calibre/srv/opts.py b/src/calibre/srv/opts.py
index b9ebffe..811e624 100644
--- a/src/calibre/srv/opts.py
+++ b/src/calibre/srv/opts.py
@@ -110,8 +110,8 @@ raw_options = (
     ' there are more than this number of items. Set to zero to disable.'),
 
     _('The interface on which to listen for connections'),
-    'listen_on', '0.0.0.0',
-    _('The default is to listen on all available IPv4 interfaces. You can change this to, for'
+    'listen_on', '::',
+    _('The default is to listen on all available IPv6 and IPv4 interfaces. You can change this to, for'
     ' example, "127.0.0.1" to only listen for connections from the local machine, or'
     ' to "::" to listen to all incoming IPv6 and IPv4 connections.'),
 
