From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Tue, 21 Nov 2023 20:30:40 +0900
Subject: Use None for empty value

Forwarded: https://github.com/kovidgoyal/calibre/pull/2119
---
 src/calibre/srv/opts.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/srv/opts.py b/src/calibre/srv/opts.py
index a6a8f14..f6e3fd7 100644
--- a/src/calibre/srv/opts.py
+++ b/src/calibre/srv/opts.py
@@ -110,7 +110,7 @@ raw_options = (
     ' there are more than this number of items. Set to zero to disable.'),
 
     _('The interface on which to listen for connections'),
-    'listen_on', '',
+    'listen_on', None,
     _('The default is to listen on all available IPv6 and IPv4 interfaces. You can change this to, for'
     ' example, "127.0.0.1" to only listen for connections from the local machine, or'
     ' to "::" to listen to all incoming IPv6 and IPv4 connections.'),
