From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Tue, 12 May 2026 20:44:26 +0900
Subject: Specify exact bitmap format to reproducible build

---
 src/calibre/linux.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/linux.py b/src/calibre/linux.py
index 6c7e1eb..15587e3 100644
--- a/src/calibre/linux.py
+++ b/src/calibre/linux.py
@@ -1333,7 +1333,7 @@ def write_appdata(key, entry, base, translators):
 
 def render_img(image, dest, width=128, height=128):
     from qt.core import QImage, Qt
-    img = QImage(I(image)).scaled(int(width), int(height), Qt.AspectRatioMode.IgnoreAspectRatio, Qt.TransformationMode.SmoothTransformation)
+    img = QImage(I(image)).convertToFormat(QImage.Format.Format_ARGB32).scaled(int(width), int(height), Qt.AspectRatioMode.IgnoreAspectRatio, Qt.TransformationMode.SmoothTransformation)
     img.save(dest)
 
 
