From: Martin Pitt <mpitt@debian.org>
Date: Mon, 14 Nov 2016 22:41:23 +0100
Subject: Work around broken threading on mips builders

Bug-Debian: http://bugs.debian.org/760865
---
 setup/build_environment.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/setup/build_environment.py b/setup/build_environment.py
index b79a4e8..3d1d7de 100644
--- a/setup/build_environment.py
+++ b/setup/build_environment.py
@@ -43,6 +43,9 @@ if islinux and not PKGCONFIG:
 if iswindows:
     import win32api
     cpu_count = win32api.GetSystemInfo()[5]
+# http://bugs.debian.org/760865
+elif 'mips' in os.uname()[4]:
+    cpu_count = 1
 else:
     from multiprocessing import cpu_count
     try:
