From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 3 Sep 2021 23:01:01 +0900
Subject: "description" tag can't have "lang" attributes

---
 src/calibre/linux.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/linux.py b/src/calibre/linux.py
index 34c09c7..265532b 100644
--- a/src/calibre/linux.py
+++ b/src/calibre/linux.py
@@ -1223,7 +1223,7 @@ def make_appdata_releases():
         # Formatting of release description tries to resemble that of
         # https://calibre-ebook.com/whats-new while taking into account the limits imposed by
         # https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-description
-        description = E.description(**{'{http://www.w3.org/XML/1998/namespace}lang': 'en'})
+        description = E.description()
         if 'new features' in revision:
             description.append(E.p('New features:'))
             description.append(E.ol(
