From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 7 Apr 2023 16:50:54 +0900
Subject: Sort lines for reproducible build

Forwarded: not-needed
---
 setup/browser_data.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/browser_data.py b/setup/browser_data.py
index df9835d..a1f16c1 100644
--- a/setup/browser_data.py
+++ b/setup/browser_data.py
@@ -102,5 +102,5 @@ def get_data():
         'user_agents_popularity': ua_freq_map,
         'timestamp': datetime.utcnow().isoformat() + '+00:00',
     }
-    ans['desktop_platforms'] = list(all_desktop_platforms(ans['common_user_agents']))
+    ans['desktop_platforms'] = list(sorted(all_desktop_platforms(ans['common_user_agents'])))
     return ans
