From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Sat, 8 Apr 2023 19:48:40 +0900
Subject: Use relative path to deterministic build

Forwarded: not-needed
---
 setup/translations.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/setup/translations.py b/setup/translations.py
index c764ff4..4e92575 100644
--- a/setup/translations.py
+++ b/setup/translations.py
@@ -488,7 +488,11 @@ class Translations(POT):  # {{{
         with open(f, 'rb') as s:
             data = s.read()
         h = hashlib.sha1(data)
-        h.update(f.encode('utf-8'))
+
+        basepath = os.path.dirname(Command.SRC)
+        rel_f    = os.path.relpath(f, basepath)
+
+        h.update(rel_f.encode('utf-8'))
         return data, h
 
     def compile_content_server_translations(self):
