From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Tue, 9 Apr 2024 23:02:11 +0900
Subject: Add keywords to desktop application entry file

Lintian fix
---
 src/calibre/linux.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/calibre/linux.py b/src/calibre/linux.py
index 21468a8..71f736b 100644
--- a/src/calibre/linux.py
+++ b/src/calibre/linux.py
@@ -1099,6 +1099,7 @@ Exec=lrfviewer %f
 Icon=calibre-viewer
 MimeType=application/x-sony-bbeb;
 Categories=Office;Viewer;
+Keywords=lrf;viewer;
 '''
 
 EVIEWER = '''\
@@ -1112,6 +1113,7 @@ TryExec=ebook-viewer
 Exec=ebook-viewer %f
 Icon=calibre-viewer
 Categories=Office;Viewer;
+Keywords=epub;ebook;viewer;
 '''
 
 ETWEAK = '''\
@@ -1125,6 +1127,7 @@ TryExec=ebook-edit
 Exec=ebook-edit %f
 Icon=calibre-ebook-edit
 Categories=Office;WordProcessor
+Keywords=epub;ebook;editor;
 '''
 
 GUI = '''\
@@ -1139,6 +1142,7 @@ Exec=calibre %U
 Icon=calibre-gui
 Categories=Office;
 X-GNOME-UsesNotifications=true
+Keywords=epub;ebook;manager;
 '''
 
 
