From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Mon, 14 Aug 2023 00:01:11 +0900
Subject: Really choose oldest file modification time in target files

---
 setup/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/__init__.py b/setup/__init__.py
index 4d353a4..8288fb8 100644
--- a/setup/__init__.py
+++ b/setup/__init__.py
@@ -47,7 +47,7 @@ def newer(targets, sources):
         if not os.path.exists(f):
             return True
     ttimes = map(lambda x: os.stat(x).st_mtime, targets)
-    oldest_target = max(ttimes)
+    oldest_target = min(ttimes)
     try:
         stimes = map(lambda x: os.stat(x).st_mtime, sources)
         newest_source = max(stimes)
