From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 19 Jul 2024 13:15:59 +0900
Subject: "lrfviewer" doesn't has AppStream meta info file

---
 src/calibre/linux.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/calibre/linux.py b/src/calibre/linux.py
index a00e445..0b81495 100644
--- a/src/calibre/linux.py
+++ b/src/calibre/linux.py
@@ -1100,6 +1100,7 @@ Icon=calibre-viewer
 MimeType=application/x-sony-bbeb;
 Categories=Office;Viewer;
 Keywords=lrf;viewer;
+X-AppStream-Ignore=true
 '''
 
 EVIEWER = '''\
