From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 19 Jul 2024 02:18:47 +0900
Subject: Use "appstream-id" key for AppStream ID value

---
 src/calibre/linux.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/linux.py b/src/calibre/linux.py
index cb06846..6f8bf42 100644
--- a/src/calibre/linux.py
+++ b/src/calibre/linux.py
@@ -1287,7 +1287,7 @@ def write_appdata(key, entry, base, translators):
                 description.append(E.p(tp, **{'{http://www.w3.org/XML/1998/namespace}lang': lang}))
 
     root = E.component(
-        E.id(key + '.desktop'),
+        E.id(entry['appstream-id']),
         E.name(entry['name']),
         E.metadata_license('CC0-1.0'),
         E.project_license('GPL-3.0'),
