From: Martin Pitt <mpitt@debian.org>
Date: Mon, 14 Nov 2016 22:41:23 +0100
Subject: Disable update check by default

---
 src/calibre/gui2/main.py |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- calibre-debian.git.orig/src/calibre/gui2/main.py
+++ calibre-debian.git/src/calibre/gui2/main.py
@@ -51,8 +51,8 @@ path_to_ebook to the database.
                       help=_('Start minimized to system tray.'))
     parser.add_option('-v', '--verbose', default=0, action='count',
                       help=_('Ignored, do not use. Present only for legacy reasons'))
-    parser.add_option('--no-update-check', default=False, action='store_true',
-            help=_('Do not check for updates'))
+    parser.add_option('--update-check', dest='no_update_check', default=True, action='store_false',
+            help=_('Check for updates'))
     parser.add_option('--ignore-plugins', default=False, action='store_true',
             help=_('Ignore custom plugins, useful if you installed a plugin'
                 ' that is preventing calibre from starting'))
