Description: Work around broken threading on mips builders
Author: Martin Pitt <mpitt@debian.org>
Bug-Debian: http://bugs.debian.org/760865

Index: calibre-2.2.0+dfsg/setup/build_environment.py
===================================================================
--- calibre-2.2.0+dfsg.orig/setup/build_environment.py
+++ calibre-2.2.0+dfsg/setup/build_environment.py
@@ -42,6 +42,9 @@ PKGCONFIG = os.environ.get('PKG_CONFIG',
 if iswindows:
     import win32api
     cpu_count = win32api.GetSystemInfo()[5]
+# http://bugs.debian.org/760865
+elif 'mips' in os.uname()[4]:
+    cpu_count = 1
 else:
     from multiprocessing import cpu_count
     try:
