From 0b6e95a080548c662164890b8af6cf26b2521b74 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Mon, 18 Nov 2019 20:28:38 +0530
Subject: [PATCH 011/106] Remove some more referefences to the old lzma

---
 COPYRIGHT                | 12 ------------
 src/calibre/constants.py |  1 -
 2 files changed, 13 deletions(-)

diff --git a/COPYRIGHT b/COPYRIGHT
index 6da2e3ad7d..10afc65876 100644
--- a/COPYRIGHT
+++ b/COPYRIGHT
@@ -12,18 +12,6 @@ Files: resources/rapydscript/*
 Copyright: Various
 License: BSD
 
-Files: src/html5lib/*
-Copyright: Copyright (c) 2006-2013 James Graham and other contributors
-License: Expat
-
-Files: src/lzma/*
-Copyright: Igor Pavlov
-License: Public Domain
-
-Files: src/lzma/*.py src/lzma/lzma_binding.c
-Copyright: Kovid Goyal
-License: GPLv3
-
 Files: src/templite/*
 Copyright: Copyright (c) 2009 joonis new media, Thimo Kraemer
 License: GPL-2+
diff --git a/src/calibre/constants.py b/src/calibre/constants.py
index 9c0d8c3f6e..c349b167bf 100644
--- a/src/calibre/constants.py
+++ b/src/calibre/constants.py
@@ -182,7 +182,6 @@ class Plugins(collections.Mapping):
                 'matcher',
                 'tokenizer',
                 'certgen',
-                'lzma_binding',
             ]
         if iswindows:
             plugins.extend(['winutil', 'wpd', 'winfonts'])
-- 
2.26.0

