From e89019bc3e593784ec7b13b8d7bc222be5264130 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 3 Dec 2019 10:47:43 +0530
Subject: [PATCH 028/106] Ensure no __pycache__ is added to pylib.zip

---
 bypy/windows/__main__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bypy/windows/__main__.py b/bypy/windows/__main__.py
index c17c619279..758724d3d8 100644
--- a/bypy/windows/__main__.py
+++ b/bypy/windows/__main__.py
@@ -570,6 +570,8 @@ def build_launchers(env, debug=False):
 
 
 def add_to_zipfile(zf, name, base, zf_names):
+    if '__pycache__' in name:
+        return
     abspath = j(base, name)
     name = name.replace(os.sep, '/')
     if name in zf_names:
-- 
2.26.0

