From a873b54cc8bbaff74c5175ce317f8df01c500226 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 3 Dec 2019 10:47:43 +0530
Subject: [PATCH 028/118] Ensure no __pycache__ is added to pylib.zip

---
 bypy/windows/__main__.py |    2 ++
 1 file changed, 2 insertions(+)

Index: b/bypy/windows/__main__.py
===================================================================
--- a/bypy/windows/__main__.py
+++ b/bypy/windows/__main__.py
@@ -573,6 +573,8 @@ def build_launchers(env, debug=False):
 
 
 def add_to_zipfile(zf, name, base, zf_names):
+    if '__pycache__' in name:
+        return
     abspath = j(base, name)
     name = name.replace(os.sep, '/')
     if name in zf_names:
