From dc25ef09bc71cec9c02f47266e4978619e44b793 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 3 Dec 2019 10:50:45 +0530
Subject: [PATCH 029/112] Fix failing winutil test

---
 src/calibre/test_build.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/calibre/test_build.py b/src/calibre/test_build.py
index c8b893692c..b02deea746 100644
--- a/src/calibre/test_build.py
+++ b/src/calibre/test_build.py
@@ -163,6 +163,8 @@ class BuildTest(unittest.TestCase):
         for fmt in (fmt, fmt.encode('ascii')):
             x = strftime(fmt, t)
             au(x, 'strftime')
+            if isinstance(fmt, bytes):
+                fmt = fmt.decode('ascii')
             self.assertEqual(unicode_type(time.strftime(fmt.replace('%e', '%#d'), t)), x)
 
     def test_sqlite(self):
-- 
2.26.2

