From 6c78bb5445787c4e78417164e9c330669af5e513 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Wed, 4 Dec 2019 09:30:38 +0530
Subject: [PATCH 034/119] HRESULT has moved from wintypes in python 3.8

---
 src/calibre/gui2/__init__.py |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: b/src/calibre/gui2/__init__.py
===================================================================
--- a/src/calibre/gui2/__init__.py
+++ b/src/calibre/gui2/__init__.py
@@ -1466,12 +1466,13 @@ empty_index = empty_model.index(0)
 def set_app_uid(val):
     import ctypes
     from ctypes import wintypes
+    from ctypes import HRESULT
     try:
         AppUserModelID = ctypes.windll.shell32.SetCurrentProcessExplicitAppUserModelID
     except Exception:  # Vista has no app uids
         return False
     AppUserModelID.argtypes = [wintypes.LPCWSTR]
-    AppUserModelID.restype = wintypes.HRESULT
+    AppUserModelID.restype = HRESULT
     try:
         AppUserModelID(unicode_type(val))
     except Exception as err:
