From a2077c373fdbed68fa6c955f1a35d10b977b9809 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Thu, 5 Dec 2019 10:20:41 +0530
Subject: [PATCH 052/107] Forgot to change tarball url for unix

---
 setup/unix-ci.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/unix-ci.py b/setup/unix-ci.py
index f24cde2272..8d1cdb0dba 100644
--- a/setup/unix-ci.py
+++ b/setup/unix-ci.py
@@ -103,7 +103,7 @@ def main():
 
         tball = 'macos-64' if ismacos else 'linux-64'
         download_and_decompress(
-            'https://download.calibre-ebook.com/ci/calibre/{}.tar.xz'.format(tball), SW
+            'https://download.calibre-ebook.com/ci/calibre3/{}.tar.xz'.format(tball), SW
         )
         if not ismacos:
             run('sudo', 'apt-get', 'update', '-y')
-- 
2.25.1

