From b5f0da81c8c73bb26b1ff44677d80298c37d50fe Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Thu, 5 Dec 2019 20:45:33 +0530
Subject: [PATCH 063/423] ...

---
 src/calibre/srv/tests/loop.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/calibre/srv/tests/loop.py b/src/calibre/srv/tests/loop.py
index 6bdcfb2608..b0ca60e498 100644
--- a/src/calibre/srv/tests/loop.py
+++ b/src/calibre/srv/tests/loop.py
@@ -301,3 +301,8 @@ class LoopTest(BaseTest):
         self.assertIn('a testing error', tb)
         jm.start_job('simple test', 'calibre.srv.jobs', 'sleep_test', args=(1.0,))
         jm.shutdown(), jm.wait_for_shutdown(monotonic() + 1)
+
+
+def find_tests():
+    import unittest
+    return unittest.defaultTestLoader.loadTestsFromTestCase(LoopTest)
-- 
2.28.0

