From 64a5b7ad0f6a43dad720a3d1f6978b3dff1caf45 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Fri, 6 Dec 2019 12:00:37 +0530
Subject: [PATCH 070/106] ...

---
 setup/parallel_build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/parallel_build.py b/setup/parallel_build.py
index 9c1a0bbaa9..c93b799a3c 100644
--- a/setup/parallel_build.py
+++ b/setup/parallel_build.py
@@ -40,7 +40,7 @@ def run_worker(job, decorate=True):
 
 
 def create_job(cmd, human_text=None, cwd=None):
-    return (cmd, human_text)
+    return Job(cmd, human_text, cwd)
 
 
 def parallel_build(jobs, log, verbose=True):
-- 
2.26.0

