From 4eb99bc55db240550cd5c329e18703bd17fab5d7 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Thu, 12 Dec 2019 14:03:26 +0530
Subject: [PATCH 075/119] Bump version for beta builds

---
 src/calibre/constants.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/src/calibre/constants.py
===================================================================
--- a/src/calibre/constants.py
+++ b/src/calibre/constants.py
@@ -5,7 +5,7 @@ from polyglot.builtins import map, unico
 import sys, locale, codecs, os, importlib, collections
 
 __appname__   = 'calibre'
-numeric_version = (4, 17, 0)
+numeric_version = (4, 99, 0)
 __version__   = '.'.join(map(unicode_type, numeric_version))
 git_version   = None
 __author__    = "Kovid Goyal <kovid@kovidgoyal.net>"
