From 532bbd1a390831363907a0625daddc6f44a2ed5a Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Thu, 12 Dec 2019 14:03:26 +0530
Subject: [PATCH 075/106] Bump version for beta builds

---
 src/calibre/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/constants.py b/src/calibre/constants.py
index a0153d02a1..0329f0f672 100644
--- a/src/calibre/constants.py
+++ b/src/calibre/constants.py
@@ -5,7 +5,7 @@ from polyglot.builtins import map, unicode_type, environ_item, hasenv, getenv, a
 import sys, locale, codecs, os, importlib, collections
 
 __appname__   = 'calibre'
-numeric_version = (4, 11, 0)
+numeric_version = (4, 99, 0)
 __version__   = '.'.join(map(unicode_type, numeric_version))
 git_version   = None
 __author__    = "Kovid Goyal <kovid@kovidgoyal.net>"
-- 
2.25.1

