From 2054fdefbc100efcfbee487c7baed483b3a2f353 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Fri, 20 Dec 2019 07:38:49 +0530
Subject: [PATCH 099/119] Cant pass byte strings to WinDLL()

---
 src/calibre/gui2/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/gui2/__init__.py b/src/calibre/gui2/__init__.py
index c58262279c..499a82b7ab 100644
--- a/src/calibre/gui2/__init__.py
+++ b/src/calibre/gui2/__init__.py
@@ -1010,7 +1010,7 @@ class Application(QApplication):
         try:
             if self.clipboard().ownsClipboard():
                 import ctypes
-                ctypes.WinDLL(b'ole32.dll').OleFlushClipboard()
+                ctypes.WinDLL('ole32.dll').OleFlushClipboard()
         except Exception:
             import traceback
             traceback.print_exc()
-- 
2.26.2

