From 0103209fd6e66cc380bb787a2b7843c3566a335d Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Mon, 6 Jan 2020 17:53:27 +0530
Subject: [PATCH 099/284] Bump beta version

---
 src/calibre/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/constants.py b/src/calibre/constants.py
index 61f0b3a730..2c3d72def1 100644
--- a/src/calibre/constants.py
+++ b/src/calibre/constants.py
@@ -5,7 +5,7 @@ from polyglot.builtins import map, unicode_type, environ_item, hasenv, getenv, a
 import sys, locale, codecs, os, importlib, collections
 
 __appname__   = 'calibre'
-numeric_version = (4, 99, 2)
+numeric_version = (4, 99, 3)
 __version__   = '.'.join(map(unicode_type, numeric_version))
 git_version   = None
 __author__    = "Kovid Goyal <kovid@kovidgoyal.net>"
-- 
2.28.0

