From 42b06c538b78ce8748ea243d0578e7551ad1c1d2 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 21 Apr 2020 22:19:56 +0530
Subject: [PATCH 109/284] ...

---
 src/polyglot/plistlib.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/polyglot/plistlib.py b/src/polyglot/plistlib.py
index b07f25765c..e31b23f05f 100644
--- a/src/polyglot/plistlib.py
+++ b/src/polyglot/plistlib.py
@@ -9,3 +9,4 @@ def wrap_bytes(x):
     return x
 
 from plistlib import loads, dumps, Data  # noqa
+loads_binary_or_xml = loads
-- 
2.28.0

