From 12f37cd374e77929d0e8c1a3bf4499994df2d41f Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 21 Apr 2020 22:19:56 +0530
Subject: [PATCH 111/119] ...

---
 src/polyglot/plistlib.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/polyglot/plistlib.py b/src/polyglot/plistlib.py
index 5b1f130239..e1a5b8b16a 100644
--- a/src/polyglot/plistlib.py
+++ b/src/polyglot/plistlib.py
@@ -3,3 +3,4 @@
 # License: GPL v3 Copyright: 2019, Kovid Goyal <kovid at kovidgoyal.net>
 
 from plistlib import loads, dumps, Data  # noqa
+loads_binary_or_xml = loads
-- 
2.26.2

