From 5ac2e8c9c3f9b6edad68f320046e1b4a284f000e Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Fri, 1 May 2020 20:05:18 +0530
Subject: [PATCH 114/423] Tell MSVC our source files are UTF-8

---
 setup/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/build.py b/setup/build.py
index c4889cbe00..ef93146a56 100644
--- a/setup/build.py
+++ b/setup/build.py
@@ -200,7 +200,7 @@ def init_env():
 
     if iswindows:
         cc = cxx = msvc.cc
-        cflags = '/c /nologo /MD /W3 /EHsc /DNDEBUG'.split()
+        cflags = '/c /nologo /MD /W3 /EHsc /utf-8 /DNDEBUG'.split()
         ldflags = '/DLL /nologo /INCREMENTAL:NO /NODEFAULTLIB:libcmt.lib'.split()
         # cflags = '/c /nologo /Ox /MD /W3 /EHsc /Zi'.split()
         # ldflags = '/DLL /nologo /INCREMENTAL:NO /DEBUG'.split()
-- 
2.28.0

