From 06a4511312c3983974ed52b307451ae2212c610b Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Fri, 1 May 2020 22:12:34 +0530
Subject: [PATCH 116/125] Netter beta upload progress

---
 setup/publish.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/publish.py b/setup/publish.py
index 4409d884b2..3473d22609 100644
--- a/setup/publish.py
+++ b/setup/publish.py
@@ -122,7 +122,7 @@ class PublishBetas(Command):
     def run(self, opts):
         dist = self.a(self.j(self.d(self.SRC), 'dist'))
         subprocess.check_call((
-            'rsync --partial -rh --progress --delete-after %s/ download.calibre-ebook.com:/srv/download/betas/'
+            'rsync --partial -rh --info=progress2 --delete-after %s/ download.calibre-ebook.com:/srv/download/betas/'
             % dist
         ).split())
 
-- 
2.27.0

