From 0b411f6e329703c365200451314138abb1200a4e Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 26 May 2020 22:28:59 +0530
Subject: [PATCH 118/423] another useless __future__

---
 src/calibre/gui2/viewer/integration.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/calibre/gui2/viewer/integration.py b/src/calibre/gui2/viewer/integration.py
index 76a9911612..6252cb5979 100644
--- a/src/calibre/gui2/viewer/integration.py
+++ b/src/calibre/gui2/viewer/integration.py
@@ -2,8 +2,6 @@
 # vim:fileencoding=utf-8
 # License: GPL v3 Copyright: 2020, Kovid Goyal <kovid at kovidgoyal.net>
 
-from __future__ import absolute_import, division, print_function, unicode_literals
-
 import os
 import re
 
-- 
2.28.0

