From a95cff84b590e45b6dc415f9f95814680fea12ec Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Mon, 15 Jun 2020 08:12:47 +0530
Subject: [PATCH 123/284] Accidentally removed python 2 needed for building Qt

---
 bypy/windows.conf | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/bypy/windows.conf b/bypy/windows.conf
index 1c80890626..0bda045ae9 100644
--- a/bypy/windows.conf
+++ b/bypy/windows.conf
@@ -1,4 +1,4 @@
-# Requires installation of Visual Studio 2017 Community Edition, WiX Toolset, Git, Ruby, Python and Perl
+# Requires installation of Visual Studio 2017 Community Edition, WiX Toolset, Git, Ruby, Python (2 and 3) and Perl
 # git.exe must be in PATH. Must have ~120GB available disk space and 8GB RAM
 # Install certifi in python 3 with:
 # py.exe -m pip install certifi
@@ -12,6 +12,7 @@
 vm_name 'windows-calibre'
 root 'C:/r'
 python 'py.exe'
+python2 'C:/Python27/python.exe'
 perl 'C:/Strawberry/perl/bin/perl.exe'
 ruby 'C:/Ruby27-x64/bin/ruby.exe'
 mesa 'C:/mesa'
-- 
2.28.0

