From ba94f211c73b1455331926801b4e47812ac7bf8e Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 16 Jun 2020 10:23:48 +0530
Subject: [PATCH 125/125] Bump beta version

---
 src/calibre/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/constants.py b/src/calibre/constants.py
index 82d6557740..189c66aed8 100644
--- a/src/calibre/constants.py
+++ b/src/calibre/constants.py
@@ -5,7 +5,7 @@ from polyglot.builtins import map, unicode_type, environ_item, hasenv, getenv, a
 import sys, locale, codecs, os, importlib, collections
 
 __appname__   = 'calibre'
-numeric_version = (4, 99, 5)
+numeric_version = (4, 99, 6)
 __version__   = '.'.join(map(unicode_type, numeric_version))
 git_version   = None
 __author__    = "Kovid Goyal <kovid@kovidgoyal.net>"
-- 
2.27.0

