From fa7eb15fc25e2ec5a245dc8848e9ff195450c1c3 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 14 Jul 2020 18:12:43 +0530
Subject: [PATCH 193/201] Bump beta version

---
 src/calibre/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/constants.py b/src/calibre/constants.py
index 83b4ec7837..ef2804ed87 100644
--- a/src/calibre/constants.py
+++ b/src/calibre/constants.py
@@ -5,7 +5,7 @@ from polyglot.builtins import map, unicode_type, environ_item, hasenv, getenv, a
 import sys, locale, codecs, os, importlib, collections
 
 __appname__   = 'calibre'
-numeric_version = (4, 99, 6)
+numeric_version = (4, 99, 7)
 __version__   = '.'.join(map(unicode_type, numeric_version))
 git_version   = None
 __author__    = "Kovid Goyal <kovid@kovidgoyal.net>"
-- 
2.28.0.rc0

