From 23c621be4bfae61f2a41f23990b90f9e4da741b2 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Fri, 24 Jul 2020 16:03:15 +0530
Subject: [PATCH 235/423] ...

---
 src/pyj/select.pyj | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pyj/select.pyj b/src/pyj/select.pyj
index bca6c0001f..1866bce10a 100644
--- a/src/pyj/select.pyj
+++ b/src/pyj/select.pyj
@@ -69,7 +69,7 @@ def range_extents(start, end, in_flow_mode):
                     node = node.childNodes[r.startOffset]
                 if node.getBoundingClientRect:
                     erect = node.getBoundingClientRect()
-                    rect = {'left': erect.left, 'top': erect.top, 'height': 2}
+                    rect = {'left': erect.left, 'top': erect.top, 'right': erect.right, 'bottom': erect.bottom, 'width': erect.width, 'height': 2}
         ans.x = Math.round(rect.left)
         ans.y = Math.round(rect.top)
         ans.height = rect.height
-- 
2.28.0

