From f97d5b98bbd4019d04cdba4022cc247d7d107fb9 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Fri, 24 Jul 2020 16:39:54 +0530
Subject: [PATCH 239/284] ...

---
 src/pyj/select.pyj | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pyj/select.pyj b/src/pyj/select.pyj
index 8ddfc2748a..1c62daaadc 100644
--- a/src/pyj/select.pyj
+++ b/src/pyj/select.pyj
@@ -84,8 +84,8 @@ def range_extents(start, end, in_flow_mode):
 
     for_boundary(start, ans.start)
     for_boundary(end, ans.end)
-    ans.start.is_empty = ans.start.height > 0
-    ans.end.is_empty = ans.end.height > 0
+    ans.start.is_empty = ans.start.height <= 0
+    ans.end.is_empty = ans.end.height <= 0
     return ans
 
 
-- 
2.28.0

