From 747a0dbf730330de56b20ad8de05a3126a88f78e Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Wed, 5 Aug 2020 23:33:35 +0530
Subject: [PATCH 317/423] Use correct color for legacy annots

---
 src/pyj/read_book/highlights.pyj | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/pyj/read_book/highlights.pyj b/src/pyj/read_book/highlights.pyj
index 5ca909184e..5c04a9eba2 100644
--- a/src/pyj/read_book/highlights.pyj
+++ b/src/pyj/read_book/highlights.pyj
@@ -154,6 +154,8 @@ def highlight_style_as_css(s, is_dark, foreground):
             bg = builtin_color(s.which, is_dark)
         elif s.type is 'custom':
             bg = s.dark if is_dark else s.light
+        elif s.color:
+            fg = s.color
         set('background-color', bg or s['background-color'] or default_color(is_dark))
         set('color', fg or s.color or foreground)
 
-- 
2.28.0

