From 4e4ee9b0530d33d1e1e7ca6b27adc0e7fdc73f02 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Thu, 6 Aug 2020 12:29:01 +0530
Subject: [PATCH 324/408] Fix for tiny selection handle in some circumstances

---
 src/pyj/select.pyj | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pyj/select.pyj b/src/pyj/select.pyj
index 1ef6ade734..c3293d0333 100644
--- a/src/pyj/select.pyj
+++ b/src/pyj/select.pyj
@@ -86,6 +86,10 @@ def range_extents(start, end, in_flow_mode):
     for_boundary(end, ans.end)
     ans.start.is_empty = ans.start.height <= 0
     ans.end.is_empty = ans.end.height <= 0
+    if ans.end.height is 2 and ans.start.height > 2:
+        ans.end.height = ans.start.height
+    if ans.start.height is 2 and ans.end.height > 2:
+        ans.start.height = ans.end.height
     return ans
 
 
-- 
2.28.0

