From 240cb2f338177030661d42b619e76b422f6b6487 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Fri, 7 Aug 2020 10:37:26 +0530
Subject: [PATCH 332/423] Fix get_color_as_rgba() not working with var() colors

---
 src/pyj/book_list/theme.pyj | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pyj/book_list/theme.pyj b/src/pyj/book_list/theme.pyj
index d8e26e994e..82d894e72c 100644
--- a/src/pyj/book_list/theme.pyj
+++ b/src/pyj/book_list/theme.pyj
@@ -70,6 +70,8 @@ def set_ui_colors(is_dark_theme):
     for k in DEFAULT_COLORS:
         val = DEFAULT_COLORS[k][attr]
         s.setProperty('--calibre-color-' + k, val)
+    get_color_as_rgba.cache = {}
+    cached_color_to_rgba.cache = {}
 
 
 def css_for_variables():
@@ -92,6 +94,8 @@ def color_to_rgba(color):
     cvs = document.createElement('canvas')
     cvs.height = 1
     cvs.width = 1
+    if color.startsWith('var('):
+        color = window.getComputedStyle(document.documentElement).getPropertyValue(color[4:-1])
     ctx = cvs.getContext('2d')
     ctx.fillStyle = color
     ctx.fillRect(0, 0, 1, 1)
-- 
2.28.0

