From c1a0b87fe46a3c30ee17c0b9f07a5e5b7e534894 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Mon, 10 Aug 2020 09:32:55 +0530
Subject: [PATCH 360/408] Remove unused code

---
 src/pyj/select.pyj | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/src/pyj/select.pyj b/src/pyj/select.pyj
index 6c296555aa..503eb57b78 100644
--- a/src/pyj/select.pyj
+++ b/src/pyj/select.pyj
@@ -101,19 +101,6 @@ def selection_extents(in_flow_mode):
     return range_extents(sel.getRangeAt(0), in_flow_mode)
 
 
-def range_at_limit(invert_x, invert_y):
-    step = 10
-    for y in range(0, window.innerHeight + step, step):
-        if invert_y:
-            y = max(0, window.innerHeight - y)
-        for x in range(0, window.innerWidth + step, step):
-            if invert_x:
-                x = max(0, window.innerWidth - x)
-            r = range_from_point(x, y)
-            if r:
-                return r
-
-
 def move_end_of_selection(pos, start):
     sel = window.getSelection()
     if not sel.rangeCount:
-- 
2.28.0

