From 45ede6045b11122ec8a81e80a5c328cdb454b2ce Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Mon, 10 Aug 2020 09:54:52 +0530
Subject: [PATCH 362/423] Micro-optimization

---
 src/pyj/select.pyj | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/pyj/select.pyj b/src/pyj/select.pyj
index 5795918e0c..0f3de7a5e6 100644
--- a/src/pyj/select.pyj
+++ b/src/pyj/select.pyj
@@ -3,13 +3,14 @@
 from __python__ import bound_methods, hash_literals
 
 
-def caret_position_from_point(x, y):
-    if document.caretPositionFromPoint:
-        return document.caretPositionFromPoint(x, y)
-    r = document.caretRangeFromPoint(x, y)
-    if r:
-        return {'offsetNode': r.startContainer, 'offset': r.startOffset}
-    return None
+if document.caretPositionFromPoint:
+    caret_position_from_point = document.caretPositionFromPoint.bind(document)
+else:
+    caret_position_from_point = def(x, y):
+        r = document.caretRangeFromPoint(x, y)
+        if r:
+            return {'offsetNode': r.startContainer, 'offset': r.startOffset}
+        return None
 
 
 def word_boundary_regex():
-- 
2.28.0

