From 63d4fbf6c363ef96438955b04be6c69763b66504 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Tue, 11 Aug 2020 09:44:52 +0530
Subject: [PATCH 366/408] Bump beta version

---
 src/calibre/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/constants.py b/src/calibre/constants.py
index a7bb16b7d7..a1c76158b1 100644
--- a/src/calibre/constants.py
+++ b/src/calibre/constants.py
@@ -5,7 +5,7 @@ from polyglot.builtins import map, unicode_type, environ_item, hasenv, getenv, a
 import sys, locale, codecs, os, importlib, collections
 
 __appname__   = 'calibre'
-numeric_version = (4, 99, 10)
+numeric_version = (4, 99, 11)
 __version__   = '.'.join(map(unicode_type, numeric_version))
 git_version   = None
 __author__    = "Kovid Goyal <kovid@kovidgoyal.net>"
-- 
2.28.0

