From 95749f58422362e04b1ab8d4d2ff116e84a517f3 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Mon, 17 Aug 2020 21:15:56 +0530
Subject: [PATCH 402/408] Missed debug print

---
 src/pyj/read_book/highlights.pyj | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/pyj/read_book/highlights.pyj b/src/pyj/read_book/highlights.pyj
index db38ca6499..ad86d09143 100644
--- a/src/pyj/read_book/highlights.pyj
+++ b/src/pyj/read_book/highlights.pyj
@@ -624,7 +624,6 @@ def remove_highlight(annot_id, view, ev):
 def edit_notes(annot_id, notes, view, ev):
     entry = ev.currentTarget.closest('.highlight')
     get_text_dialog(_('Notes'), def(ok, text):
-        print(11111111, ok, text)
         if not ok:
             return
         text = text or ''
-- 
2.28.0

