From e5c531d22bb1ad6b593fdb2b309487f8cff48726 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Mon, 17 Aug 2020 21:33:27 +0530
Subject: [PATCH 405/423] On closing highlights panel go back to text

---
 src/pyj/read_book/overlay.pyj | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/pyj/read_book/overlay.pyj b/src/pyj/read_book/overlay.pyj
index 2d52875f75..d551818232 100644
--- a/src/pyj/read_book/overlay.pyj
+++ b/src/pyj/read_book/overlay.pyj
@@ -675,6 +675,7 @@ class Overlay:
         self.view.get_current_cfi('show-bookmarks', do_it)
 
     def show_highlights(self):
+        self.hide_current_panel()
         if ui_operations.toggle_highlights:
             self.hide()
             ui_operations.toggle_highlights()
-- 
2.28.0

