From bc583de0a2103a3ebc401f603bd0e3fc59ae92c0 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Sat, 22 Aug 2020 14:05:44 +0530
Subject: [PATCH 427/428] forkserver is not available on windows

---
 src/calibre/test_build.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/calibre/test_build.py b/src/calibre/test_build.py
index df7ee0234f..935a122080 100644
--- a/src/calibre/test_build.py
+++ b/src/calibre/test_build.py
@@ -341,8 +341,10 @@ class BuildTest(unittest.TestCase):
 
 
 def test_multiprocessing():
-    from multiprocessing import get_context
-    for stype in ('spawn', 'forkserver'):
+    from multiprocessing import get_context, get_all_start_methods
+    for stype in get_all_start_methods():
+        if stype == 'fork':
+            continue
         ctx = get_context(stype)
         q = ctx.Queue()
         arg = 'hello'
-- 
2.28.0

